src/wui/interactive_gamebase.cc

```diff
@@ -299,16 +311,54 @@ void InteractiveGameBase::gamespeed_menu_selected(GameSpeedEntry entry) {
 	}
 }
 
-void InteractiveGameBase::increase_gamespeed() {
+void InteractiveGameBase::increase_gamespeed(uint32_t speed) {
 	if (GameController* const ctrl = get_game()->game_controller()) {
-		ctrl->set_desired_speed(ctrl->desired_speed() + 1000);
+		uint32_t const current_speed = ctrl->desired_speed();
+		switch (speed) {
+		case 250:
+			ctrl->set_desired_speed(current_speed + speed);
+			break;
+		case 1000:
+			if (current_speed < 1000) {
+				ctrl->set_desired_speed(1000);
+			} else {
+				ctrl->set_desired_speed(current_speed - current_speed % speed + speed);
+			}
+			break;
+		case 10000:
+			ctrl->set_desired_speed(
+			   current_speed >= speed ? current_speed - current_speed % speed + speed : speed);
```

Comment:
What does this line do? Please add () so it´s apparent which calculations the % operator refers to, and add a comment explaining the maths here.

Reply:
Since I'm going to make it increase 1x and 10x regardless of the previous values (regarding your testing) These calculations aren't necessary anymore.